### ISSUE TITLE
Getters vs extractors: where should we draw the line?

### ISSUE THREAD

**Issue by mateuszrzeszutek** [open-telemetry/opentelemetry-java-instrumentation#8246]
Linking the original discussion: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8131#discussion_r1157375072

**Comment by breedx-splk** [comment]
Have your thoughts on this changed at all in the last 4+ months? What do you wanna do about this issue?

**Comment by mateuszrzeszutek** [comment]
> Have your thoughts on this changed at all in the last 4+ months?

Not really. I still think that we should encapsulate the bulk of the semconv logic in the extractors, and the getter should just map our concepts to instrumented library concepts.
@trask WDYT?

**Comment by trask** [comment]
from @mateuszrzeszutek's https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8131#discussion_r1157375072

> put as much as possible into extractors, and make implementing the getters as easy as possible

I think this is more or less what we are doing
